packages/playwright-test/src/runner/watchMode.ts

```diff
@@ -167,11 +184,16 @@ Waiting for file changes. Press ${colors.bold('h')} for help or ${colors.bold('q
     }
 
     if (command === 'exit')
-      return 'passed';
+      break;
 
-    if (command === 'interrupted')
-      return 'interrupted';
+    if (command === 'interrupted') {
+      result = 'interrupted';
+      break;
+    }
   }
+
+  await globalCleanup();
```

Comment:
- Maybe same `status` issue here?
- Perhaps we should create another reporter to report teardown's output/errors?